### PR TITLE
added CS50 IDE > Adminer

### PR DESCRIPTION
**Pull request by kzidane**

### CODE REVIEW

**Comment by kzidane** [comment]
menu

File: simple.js [hunk 1]
```diff
@@ -1071,6 +1078,48 @@ define(function(require, exports, module) {
...
+         */
+        function startAdminer() {
+
+            // open new browser tab
+            var tab = window.open("", "_blank");
+            if (!tab)
+                return;
+
+            tab.document.write(
+                'Starting adminer...<br>' +
```
Comment (dmalan) [inline simple.js]: `Adminer`

File: simple.js [hunk 2]
```diff
@@ -1071,6 +1078,48 @@ define(function(require, exports, module) {
...
+                'Please wait! This page will reload automatically.'
+            );
+
+            proc.spawn("adminer", { args: [ "--url-only" ] }, function(err, process) {
```
Comment (dmalan) [inline simple.js]: Is it C9 style to have spaces inside arrays?

File: simple.js [hunk 3]
```diff
@@ -1071,6 +1078,48 @@ define(function(require, exports, module) {
...
+                if (err) {
+                    console.error(err);
+                    return tab.document.write("Could not start adminer.");
```
Comment (dmalan) [inline simple.js]: Lowercase okay here, since CLI name.

File: simple.js [hunk 4]
```diff
@@ -1071,6 +1078,48 @@ define(function(require, exports, module) {
...
+                }
+
+                // survive reloads
+                process.unref();
+
+                // read url from stdout
+                var url = "";
+                process.stdout.on("data", function handleOutput(chunk) {
```
Comment (dmalan) [inline simple.js]: Is it possible a partial URL might be read via this process (and thus be broken)?

File: simple.js (hunk 2 above)
Comment (kzidane) [reply simple.js]: I thought it was, but maybe I saw that somewhere else.

File: simple.js (hunk 1 above)
Comment (kzidane) [reply simple.js]: Done.

File: simple.js (hunk 3 above)
Comment (kzidane) [reply simple.js]: k.

File: simple.js (hunk 4 above)
Comment (kzidane) [reply simple.js]: Catching a whitespace in the end for that reason https://github.com/cs50/harvard.cs50.simple/pull/105/files/5a1f2ec4c1f1a568ca8e2ad4d972e6fda67857d4#diff-b6a68d61529250580f3e0a773a626b8eR1110.

**Pull request merged**